docs/semgrep-ci/configuration-reference.md

```diff
@@ -22,20 +22,27 @@ SEMGREP_RULES="p/security-audit p/secrets"
 
 ## Diff-aware scanning (`SEMGREP_BASELINE_REF`)
 
-For [diff-aware scans](overview.md#features), set this variable
-to the git ref (branch name, tag, or commit hash) to use as a baseline.
-For example, to report findings newly added
-since branching off from your `main` branch, set
+For [diff-aware scans](overview.md#features), set this variable to create a baseline. This option limits scan results to those introduced after the git commit, in a branch, or tag. For example, you have a repository with 10 commits. You set the commit number 8 as the baseline. Consequently, Semgrep only returns scan results introduced by changes in commits 9 and 10.
 
-```sh
-SEMGREP_BASELINE_REF=main
-```
+NOTE: It is best to perform diff scans on branches other than your `main` branch. Performing a diff scan after a full-project scan of the same branch closes findings of the full-project. All findings of a particular branch are marked as fixed when the branch is re-scanned. As a consequence, Semgrep no longer reports their findings.
```

Comment:
I think this needs a lot more context, such as:

The Semgrep App keeps track of which findings have been fixed on a given branch. If you configure diff scans on your `main` branch comparing the latest `main` commit to the previous commit, the app will mistakenly consider all findings except those added in the latest commit to have been fixed.

Reply:
I see you didn't actually write this line yourself (you just moved it around), but we might as well fix it now!

Reply:
Changing diff scan to baseline scan, so we are using consistent naming. @underyx so what about this:

NOTE: It is recommended to perform baseline scans on branches other than your `main` branch. The Semgrep App keeps track of which findings have been fixed on a given branch. If you configure baseline scans on your `main` branch comparing the latest `main` commit to the penultimate commit, Semgrep mistakenly considers all findings except those added in the latest commit as fixed.

Reply:
yep, sounds good!

Reply:
Thanks! Resolving.